src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -338,6 +469,16 @@ muse::Ret Au3Interaction::pasteFromClipboard(secs_t begin, TrackId destinationTr
     // check if copied data fits into selected area
     auto ret = canPasteClips(dstTracksIds, begin);
```

Comment:
I see that canPasteClips has the same problem as makeRoomForDataOnTracks 
https://github.com/audacity/audacity/pull/7601/files#r1810845472

I would advise to get data (std::vector<TrackData>) from clipboard as early as possible and pass it to methods (rather than calling clipboard).

Well, we need to solve the problem with assumptions.